servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -399,7 +406,13 @@ private Single<HttpServerContext> doBind(final HttpExecutionContext executionCon
         ReadOnlyHttpServerConfig roConfig = config.asReadOnly();
         StreamingHttpService filteredService = applyInternalFilters(service, roConfig.lifecycleObserver());
 
-        if (roConfig.tcpConfig().isAlpnConfigured()) {
+        // Important: This code needs to be kept in-sync with the similar, but not identical
+        // code inside OptionalSslNegotiator#bind
+        if (roConfig.tcpConfig().sslConfig() != null &&
+                roConfig.tcpConfig().sslListenMode() == SslListenMode.SSL_OPTIONAL) {
+            return OptionalSslNegotiator.bind(executionContext, config, address, connectionAcceptor, filteredService,
```

Comment:
Instead of doing refactoring of `ReadOnlyTcpServerConfig`, consider following an approach we have for `HttpClientConfig`. It has a copy constructor, that invokes a copy ctor of `TcpClientConfig` and `HttpConfig`.

If we add similar copy ctors for `HttpServerConfig` and `TcpServerConfig`, we will be able to make a full copy before invoking `OptionalSslNegotiator.bind`, then reset `SslConfig` (we can make `AbstractTcpConfig.sslConfig(...) param `@Nullable`), and create a read-only version without `SslConfig` upfront. Both configs will be passed to `bind`. A config copy will be created only once per server, not per each new connection.